Release/4.0 (#556)
SimpleDataLabsInc/prophecy-docs: docs/analysts/development/execution/execution.md

---
title: Execution
id: execution
description: Run a set of defined operations in a pipeline
tags: []
---

> In Prophecy, execution refers to running a set of defined operations in a model or pipeline. Pipelines in SQL projects can utilize a few resources for computation:

Author: I can rewrite the first sentence if it's confusing. Technically it's not an "execution" that occurs but rather pipeline execution=the running of a pipeline. The user executes the pipeline.

- **Prophecy Runtime**: handles the flow of data in and out of the pipeline.

> - **SQL Warehouse**: executes the data transformations.
> 
> When you create a [Prophecy fabric](/administration/prophecy-fabrics/), you can choose which SQL warehouse to use for pipeline execution.

Reviewer: ```suggestion
- **SQL Warehouse**: executes the data transformations. 
   You can select which SQL warehouse to use for pipeline execution when you create a [Prophecy fabric](/administration/prophecy-fabrics/).
```

Suggestion: I think you should append/indent this sentence to the SQL warehouse bullet because when I read it the first time, I was trying to figure out what it had to do with the page.

Author: True. I will include it in the second bullet

## Interactive execution

> Running your pipeline during development can help you understand how your data is being transformed. For instance, when you configure a new gem in the pipeline, you might want to inspect its output. You can produce this data sample in two ways:

Reviewer: ```suggestion
You can understand how Prophecy transforms your data by running your pipeline during development. For example, when you configure a new gem in the pipeline, you might want to inspect its output. You can inspect the out in two ways:
```

Author: I like the first changes, but the last sentence misses just a small nuance. You inspect the output in one way: opening the data sample. You produce the data sample in two ways: the ways I listed. So I will reword it to make that clearer

- Run the entire pipeline using the **play** button on the pipeline canvas.
- Execute the pipeline up to and including a particular gem using the **play** button on that gem.

When you click on an interim data sample, you load the data and open the [Data Explorer](docs/analysts/development/data-explorer.md). The Data Explorer lets you sort, filter, and search through the gem output.

## External sources

> When you read from and write to external sources, certain steps happen in the backend during pipeline runs to make the data compatible with SQL. Because SQL requires [tables](/analysts/development/gems/source-target/#tables) to perform transformations, the Prophecy runtime will automatically create temporary tables in your SQL warehouse throughout the pipeline when necessary. These tables allow SQL to process external data and transform it.

Reviewer: ```suggestion
When you read from and write to external sources, your pipeline runs steps behind the scenes to make the data compatible with SQL. Because SQL requires [tables](/analysts/development/gems/source-target/#tables) to perform transformations, the Prophecy runtime will automatically create temporary tables in your SQL warehouse throughout the pipeline when necessary. These tables allow SQL to process external data and transform it.
```
